packages/tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,285 @@
+
+  /**
+   * Callback to be executed when a tab is selected.
+   */
+  onChange?: React.ReactEventHandler;
+
+  /**
+   * Tab that should appear active. If value passed to selected prop, component will be controlled by consumer.
+   */
+  selected?: string | null;
+
+  /**
+   * className supplied to Tabs container.
+   */
+  className?: string;
+}
+
+/**
+ * # Tabs
+ *
+ * Tabs component
+ *
+ * ```
+<Tabs onChange={() => execute callback onChange}>
+  <Tab value='tab-1' title='First Tab'>Tab 1</Tab>
+  <Tab value='tab-2' title='Second Tab'>Tab 2</Tab>
+</Tabs>
+```
+ * @param props.children Content to appear inside of Tabs component.
+ * @param props.onChange Callback to be executed when a Tab is selected.
+ * @param props.selected Tab that should appear active. If value passed, component will be controlled by consumer.
+ * @param props.className Classname applied to Tabs container.
+ */
+function Tabs({ children, onChange, selected, className, ...rest }: TabsProps) {
+  const [activeTab, setActiveTab] = useState();
+  const [currentIndex, setCurrentIndex] = useState();
+  const [focusedState, setFocusedState] = useState([] as Array<string>);
+  const listRef = useRef<HTMLUListElement>(null);
+
+  useEffect(() => {
+    const currIdx = React.Children.map(
+      children,
+      (child: React.ReactElement, index) => {
+        if (child === null || child === undefined) {
+          return;
+        }
+
+        if (selected || activeTab) {
+          return child.props.value === activeTab ||
+            child.props.value === selected
+            ? index
+            : null;
+        }
+
+        if (child.props.default) {
+          return index;
+        }
+
+        return null;
+      },
+    ).filter(index => index !== null);
+
+    setCurrentIndex(currIdx[0]);
+  });
+
+  function handleChange(e: React.SyntheticEvent<Element, MouseEvent>) {
+    if (!selected) {
+      setActiveTab((e.target as HTMLLIElement).getAttribute('data-tab-id'));
+    }
+
+    if (onChange) {
+      onChange(e);
+    }
+  }
+
+  function handleKeyDown(e: React.KeyboardEvent) {
+    if (selected) {
+      return;
+    }
+
+    const enabledIndexes = React.Children.toArray(children).reduce(
+      (acc, child, index) => {
+        if (child.props.disabled) {
+          return acc;
+        }
+
+        return [...acc, index];
+      },
+      [] as Array<number>,
+    );
+
+    const enabledCurrentIndex = enabledIndexes.indexOf(currentIndex);
+
+    let idx: number;
+
+    switch (e.key) {
+      case 'ArrowRight':
+      case 'ArrowDown':
+        idx = (enabledCurrentIndex + 1) % enabledIndexes.length;
+        setActiveTab(children[enabledIndexes[idx]].props.value);
+        break;
+
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        idx =
+          (enabledCurrentIndex - 1 + enabledIndexes.length) %
+          enabledIndexes.length;
+        setActiveTab(children[enabledIndexes[idx]].props.value);
+        break;
+    }
+  }
+
+  function isTab(
```

Comment:
This can be declared outside the component render function.

Reply:
👍